### ISSUE TITLE
Soho Datagrid: The checkbox column header can only be activated with the mouse but not the keyboard.

### ISSUE THREAD

**Issue by JBismonte** [infor-design/enterprise#4859]
**Describe the bug**
The checkbox column header can only be activated with the mouse. The Spacebar key does not toggle the state of this checkbox. 

**To Reproduce**

1. go to https://master-enterprise.demo.design.infor.com/components/datagrid/example-mixed-selection.html
2. navigate to the checkbox column header via keyboard and click spacebar/enter. it the checkbox's state doesnt change.

**Expected behavior**
[VPAT Recommendation]
Ensure that all interactive functionality is operable with the keyboard. Users should be able to toggle checkboxes with the Spacebar. 

**Version**

- ids-enterprise: v4.38.0

**Screenshots**

**Platform**
 - Infor Application/Team Name: Landmark Client
 - Device: (if applicable): NA
 - OS Version: Windows 10
 - Browser Name: Firefox
 - Browser Version: v85.0

**Additional context**
https://jira.lawson.com/browse/LMCLIENT-34405

**Comment by tmcconechy** [comment]
I think this did used to work

**Comment by janahintal** [comment]
QA Passed. Verified in https://master-enterprise.demo.design.infor.com/components/datagrid/example-mixed-selection.html
Able to select All rows using key board functions but not for individual selection.
